File: GameLibrary/src/cc/lib/checkerboard/CanadianDraughts.java
package cc.lib.checkerboard;

import cc.lib.game.Utils;

import static cc.lib.checkerboard.Game.FAR;
import static cc.lib.checkerboard.Game.NEAR;
import static cc.lib.checkerboard.PieceType.CHECKER;
import static cc.lib.checkerboard.PieceType.EMPTY;

public class CanadianDraughts extends Draughts {

    @Override
    public void init(Game game) {
        game.init(12, 12);
        game.initRank(0, FAR, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY);
        game.initRank(1, FAR, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER);
        game.initRank(2, FAR, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY);
        game.initRank(3, FAR, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER);
        game.initRank(2, FAR, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY);
        game.initRank(4, -1, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY);
        game.initRank(5, -1, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY);
        game.initRank(7, NEAR, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER);
        game.initRank(6, NEAR, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY);
        game.initRank(7, NEAR, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER);
        game.initRank(8, NEAR, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY);
        game.initRank(9, NEAR, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER, EMPTY, CHECKER);
        game.setTurn(Utils.flipCoin() ? FAR : NEAR);
    }
}

File: Checkerboard/src/cc/lib/checkerboard/CanadianDraughts.kt
package cc.lib.checkerboard

import cc.lib.game.Utils

class CanadianDraughts : Draughts() {
	override fun init(game: Game): Array<Array<Piece>> {
		game.turn = if (Utils.flipCoin()) Game.FAR else Game.NEAR
		return initFromPieceTypes(arrayOf(
			Game.FAR to arrayOf(PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY),
			Game.FAR to arrayOf(PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER),
			Game.FAR to arrayOf(PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY),
			Game.FAR to arrayOf(PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER),
			Game.FAR to arrayOf(PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY),
			Game.NOP to arrayOf(PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY),
			Game.NOP to arrayOf(PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY, PieceType.EMPTY),
			Game.NEAR to arrayOf(PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER),
			Game.NEAR to arrayOf(PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY),
			Game.NEAR to arrayOf(PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER),
			Game.NEAR to arrayOf(PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY),
			Game.NEAR to arrayOf(PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER, PieceType.EMPTY, PieceType.CHECKER),
		))
	}
}